compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -55,9 +55,12 @@ struct LambdaContext {
 struct Monomorphizer<'interner> {
     /// Functions are keyed by their unique ID and expected type so that we can monomorphize
     /// a new version of the function for each type.
+    /// We also key by any turbofish generics that are specified.
+    /// This is necessary for a case where we may have a trait generic that can be instantiated
+    /// outside of a function parameter or return value.
```

Comment:
Can you elaborate on the justification here? Why does the monomorphizer need to handle turbofish generics separately at all? I'd think that the turbofish list would just be the stored as the instantiation list.